### PR TITLE
Phone book

### PR DESCRIPTION
**Pull request by horidon**

### CODE REVIEW

**Review by OlegLustenko — CHANGES_REQUESTED** [review]

File: phone-book/src/modules/keypad.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                ${this.createCallOffButtonAndErrorBlock()}
+            </main>
+        `;
+
+        return keypadPanel;
+    }
+
+    createArrayButtons() {
+        return buttonValues.reduce((newArr, elem) => {
+            const buttonForKeypad = createElem('div');
```
[the remark this answers is not in the document]
Comment (Balashov-nikita) [reply phone-book/src/modules/keypad.js]: by hand(((

File: phone-book/src/modules/helper.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class HelpfulFeaturesForElementPrototype{
+    constructor() {
+
+        Element.prototype.removeClass = function(className) {
```
[the remark this answers is not in the document]
Comment (Balashov-nikita) [reply phone-book/src/modules/helper.js]: why not?

File: phone-book/src/modules/call-mode.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import {
+    HelpfulFeaturesForElementPrototype,
+    createElem
```
Comment (OlegLustenko) [inline phone-book/src/modules/call-mode.js]: I don't see usage of createElem

File: phone-book/src/modules/contact.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    applyListeners() {
+        const titlesForContactList = document.getElementById('wraper-for-th');
+
+        titlesForContactList.addEventListener('click', (e) => {
+            const listOfContacts = document.getElementById('list-of-contacts');
+
+            const arrayWithUsersOnCurrecntPage = [...listOfContacts.children].reduce((newArr, user) => {
+                const userInfo = {
```
Comment (OlegLustenko) [inline phone-book/src/modules/contact.js]: you don't have to acces to real DOM, please sort your virtual-data

File: phone-book/src/modules/contact.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                </tr>
+            </thead>
+        `
+    }
+
+    contactListComponent(arrayWithUsers) {
+        const contactList = createElem('tbody');
+        contactList.id = 'list-of-contacts';
+
+        arrayWithUsers.forEach(user => {
```
Comment (OlegLustenko) [inline phone-book/src/modules/contact.js]:
```js
return `
 <tbody>
  ${this.renderUsers()}
</tbody>
`
```

**Review by OlegLustenko — CHANGES_REQUESTED** [review]

File: phone-book/src/index.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+const helper = new HelpfulFeaturesForElementPrototype();
+
+const contactPage = new ContactMode();
+
+const footer = new FooterNavigationBar();
+
+const keypad = new KeypadMode();
+
+class App{
+    constructor() {}
```
Comment (OlegLustenko) [inline phone-book/src/index.js]: please add a state to App.

**Review by OlegLustenko — DISMISSED** [review]

**Pull request closed without merging**